### PR TITLE
Sleep check rework

### PR DESCRIPTION
**Pull request by keszybz**
An attempt to make #14270 work.

### CODE REVIEW

File: src/shared/sleep-config.c [hunk 1]
```diff
@@ -204,15 +203,25 @@ static int swap_device_to_major_minor(const SwapEntry *swap, char **ret) {
         if (r < 0)
                 return log_debug_errno(errno, "Unable to stat %s: %m", swap->device);
 
-        swap_dev = streq(swap->type, "partition") ? sb.st_rdev : sb.st_dev;
-        if (asprintf(&major_minor, "%u:%u", major(swap_dev), minor(swap_dev)) < 0)
-                return log_oom();
+        if (streq(swap->type, "partition")) {
+                if(!S_ISBLK(sb.st_mode))
```
Comment (poettering) [inline src/shared/sleep-config.c]: space missing

File: src/shared/sleep-config.c [hunk 2]
```diff
@@ -368,8 +395,8 @@ int find_hibernate_location(HibernateLocation **ret_hibernate_location) {
                     || ((swap->priority == hibernate_location->swap->priority)
                         && (swap->size - swap->used) > (hibernate_location->swap->size - hibernate_location->swap->used))) {
 
-                        _cleanup_free_ char *swap_device_id = NULL;
-                        r = swap_device_to_major_minor(swap, &swap_device_id);
+                        dev_t swap_device;
+                        r = swap_device_to_device_id(swap, &swap_device);
                         if (r < 0)
```
Comment (poettering) [inline src/shared/sleep-config.c]: something should log if this call fails

Comment (keszybz) [reply src/shared/sleep-config.c]: Updated. Now swap_device_to_device_id() doesn't log at all, and the caller logs.

**Comment by poettering** [comment]
CI failures are unrelated. Merging.

**Pull request merged**